src/Magnum/ImGuiIntegration/Integration.h

```diff
@@ -0,0 +1,211 @@
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ef Magnum::ImGuiIntegration::Context
+ */
+
+#include <Magnum/GL/AbstractShaderProgram.h>
+#include <Magnum/Math/Matrix4.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/Buffer.h>
+#include <Magnum/GL/Mesh.h>
+#include <Magnum/Timeline.h>
+#include <Magnum/GL/Attribute.h>
+
+#include "Magnum/ImGuiIntegration/visibility.h"
+
+namespace Magnum { namespace ImGuiIntegration {
+
+namespace Implementation {
+
+class ImGuiShader : public GL::AbstractShaderProgram {
+    public:
+        typedef GL::Attribute<0, Vector2> Position;
+        typedef GL::Attribute<1, Vector2> TextureCoordinates;
+        typedef GL::Attribute<2, Vector4> Color;
+
+        explicit ImGuiShader();
+
+        ImGuiShader& setProjectionMatrix(const Matrix4& matrix) {
+            setUniform(_projMatrixUniform, matrix);
+            return *this;
+        }
+
+        ImGuiShader& bindTexture(GL::Texture2D& texture) {
+            texture.bind(TextureLayer);
+            return *this;
+        }
+
+    private:
+        enum: Int {
+            TextureLayer = 0
+        };
+
+        Int _projMatrixUniform;
+};
+
+}
+
+/*
+@brief ImGui context
+
+Handles initialization and destruction of ImGui context.
+
+Use @ref newFrame() to initialize a ImGui frame and finally draw it with
+@ref drawFrame() to the currently bound framebuffer.
+
+@section ImGuiIntegration-Context-usage Usage
+
+There should always only be one instance of @ref Context. As soon as this
+one instance is created, you can access it via @ref Context::get().
+
+Example:
+
+@snippet ImGuiIntegration.cpp Context-usage
+
+@section ImGuiIntegration-Context-rendering Rendering
+
+ImGui requires @ref GL::Renderer::Feature::ScissorTest "scissor test"
+to be enabled and @ref GL::Renderer::Feature::DepthTest "depth test"
+to be disabled.
+
+@ref GL::Renderer::Feature::DepthTest "Blending" should be enabled with
+and set up as in the following snippet:
+
+@snippet ImGuiIntegration.cpp Context-usage-per-frame
+*/
+class MAGNUM_IMGUIINTEGRATION_EXPORT Context {
+    public:
+
+        explicit Context();
+
+        /** @brief Copying is not allowed */
+        Context(const Context&) = delete;
+
+        /** @brief Moving is not allowed */
+        Context(Context&&) = delete;
+
+        ~Context();
+
+        /** @brief Copying is not allowed */
+        Context& operator=(const Context&) = delete;
+
+        /** @brief Moving is not allowed */
+        Context& operator=(Context&&) = delete;
+
+        /**
+         * @brief Global context instance
+         *
+         * Expects that an instance has been created.
+         */
+        static Context& get();
+
+        /**
+         * @brief Start a new frame
+         * @param winSize Window size
+         * @param viewportSize Viewport size
```

Comment:
Can youvertically align the parameter descriptions like it's done in Magnum itself? Preferrably to some tab-stop. Similarly elsewhere and in `Widgets.h`. Thanks!